provision/pull-images.sh

```diff
@@ -19,44 +19,43 @@ if [ -z "${NAME_PREFIX}" ]; then
         k8s.gcr.io/coredns:1.3.1 \
         k8s.gcr.io/coredns:1.6.2 \
         k8s.gcr.io/coredns:1.6.5 \
-        docker.io/cilium/cc-grpc-demo:v2.0 \
+        docker.io/cilium/cc-grpc-demo:v3.0 \
         docker.io/cilium/demo-client:latest \
         docker.io/cilium/demo-httpd:latest \
         docker.io/cilium/echoserver:1.10.1 \
         docker.io/cilium/echoserver-udp:v2020.01.30 \
-        docker.io/cilium/istio_pilot:1.5.4 \
-        docker.io/cilium/istio_proxy:1.5.4 \
-        docker.io/cilium/json-mock:1.1 \
+        docker.io/cilium/istio_pilot:1.5.7 \
+        docker.io/cilium/istio_proxy:1.5.7 \
+        docker.io/cilium/json-mock:1.2 \
         docker.io/cilium/kafkaclient2:latest \
         docker.io/cilium/kafkaclient:latest \
         docker.io/cilium/log-gatherer:v1.0 \
         docker.io/cilium/migrate-svc-test:v0.0.1 \
         docker.io/cilium/starwars:v1.0 \
-        docker.io/cilium/python-bmemcached:v0.0.1 \
-        docker.io/cilium/dnssec-client:v0.1 \
+        docker.io/cilium/python-bmemcached:v0.0.2 \
+        docker.io/cilium/dnssec-client:v0.2 \
         docker.io/cilium/docker-bind:v0.3 \
         docker.io/digitalwonderland/zookeeper:latest \
-        docker.io/istio/examples-bookinfo-details-v1:1.6.0 \
+        docker.io/istio/examples-bookinfo-details-v1:1.16.2 \
         docker.io/istio/examples-bookinfo-productpage-v1:0.2.3 \
-        docker.io/istio/examples-bookinfo-ratings-v1:1.6.0 \
+        docker.io/istio/examples-bookinfo-ratings-v1:1.16.2 \
         docker.io/istio/examples-bookinfo-reviews-v1:1.6.0 \
         docker.io/istio/examples-bookinfo-reviews-v2:1.6.0 \
         docker.io/library/alpine:3.9 \
         docker.io/library/cassandra:3.11.3 \
-        docker.io/library/httpd:2.4.34 \
```

Comment:
For reviewers: I removed this one because it is outdated and only used for the Documentation image.